src/main/java/org/apache/pulsar/io/jcloud/BlobStoreAbstractConfig.java

```diff
@@ -142,6 +143,11 @@ public void validate() {
             checkArgument(!StringUtils.startsWith(bytesFormatTypeSeparator, "0x"),
                     "bytesFormatTypeSeparator should be a hex encoded string, which starts with '0x'");
         }
+
+        if (pendingQueueSize == -1) {
+            pendingQueueSize = batchSize * 2;
```

Comment:
The default value might be too small and affect the performance due to frequent offload.
Did you test the performance with high throughput?

Reply:
Thanks for the comment, I have tested with high throughput with large message payload size(each message > 1MB), and it works fine on SN cloud. But havnt tested with small message payload case yet, and it seems the default value is not enough for small payload cases. So I have changed the default value to `batchSize * 10`, and people should customize the value in different cases.